kernel/wait_queue/src/lib.rs

```diff
@@ -236,3 +236,9 @@ impl WaitQueue {
         }
     }
 }
+
+impl Default for WaitQueue {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
why is this needed? If it's not, please remove.

Reply:
Clippy recommended it

Reply:
is it actually used somewhere? I don't want to add or derive a bunch of unnecessary trait impls until they are actually needed.

Reply:
If this is the only place where the lint triggers, I suppose it's fine to add it. But if this is a frequent pattern, I'd rather just disable the entire lint. Code isn't free, this style lint is rather silly imo.